src/extension/monitor/core/src/main/java/org/geoserver/monitor/MonitorFilter.java

```diff
@@ -184,7 +184,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             data.setResourcesProcessingTime(renderingTimeLayers);
             if (data.getEndTime() == null) data.setEndTime(new Date());
         }
-        monitor.update();
+        // monitor.update();
```

Comment:
I'm making a small assumption that any implementation which implements 'requestUpdated' would also implement 'requestedCompleted'.  

If that assumption is reasonable, then it is undesirable to call update() here since complete() is called two lines later.

Reply:
A lifecycle callback is meant to be generic, does not require one to implement specific methods, if you would be absurd to require so.